src/hotspot/share/oops/methodData.cpp

```diff
@@ -846,7 +855,7 @@ bool FailedSpeculation::add_failed_speculation(nmethod* nm, FailedSpeculation**
   guarantee(is_aligned(fs, sizeof(FailedSpeculation*)), "FailedSpeculation objects must be pointer aligned");
   guarantee_failed_speculations_alive(nm, failed_speculations_address);
 
-  FailedSpeculation** cursor = failed_speculations_address;
+  cursor = failed_speculations_address;
```

Comment:
Why not continue from `cursor` value from previous check loop?
Can the list be modified by other threads in between?